sdk/python/lib/pulumi/runtime/invoke.py

```diff
@@ -237,11 +239,12 @@ def do_rpc_call():
                 from ..resource import DependencyResource  # pylint: disable=import-outside-toplevel
                 deps = set(map(DependencyResource, deps_urns))
 
-                # If typ is not None, call translate_output_properties to instantiate any output types.
-                value = rpc.translate_output_properties(deserialized, lambda prop: prop, typ) if typ else deserialized
+                if is_known:
+                    # If typ is not None, call translate_output_properties to instantiate any output types.
+                    value = rpc.translate_output_properties(deserialized, lambda p: p, typ) if typ else deserialized
 
             resolve_value.set_result(value)
-            resolve_is_known.set_result(True)
+            resolve_is_known.set_result(is_known)
```

Comment:
I feel we're coding against a very low-level interface here so it sounds like it was an easy mistake to make.